src/eko/couplings.py

```diff
@@ -357,46 +377,50 @@ class Couplings:
 
     def __init__(
         self,
-        couplings_ref,
-        scale_ref,
-        masses,
-        thresholds_ratios,
-        order=(1, 0),
-        method="exact",
-        nf_ref=None,
-        max_nf=None,
-        hqm_scheme="POLE",
+        couplings: CouplingsRef,
+        order: Order,
+        method: CouplingEvolutionMethod,
+        masses: HeavyQuarkMasses,
+        hqm_scheme: QuarkMassSchemes = QuarkMassSchemes.POLE,
+        thresholds_ratios: MatchingScales = MatchingScales(c=1.0, b=1.0, t=1.0),
     ):
         # Sanity checks
-        if couplings_ref[0] <= 0:
-            raise ValueError(f"alpha_s_ref has to be positive - got {couplings_ref[0]}")
-        if couplings_ref[1] <= 0:
-            raise ValueError(
-                f"alpha_em_ref has to be positive - got {couplings_ref[1]}"
-            )
-        if scale_ref <= 0:
-            raise ValueError(f"scale_ref has to be positive - got {scale_ref}")
+        def positive(name, var):
+            if var <= 0:
+                raise ValueError(f"{name} has to be positive - got: {var}")
+
+        positive("alpha_s_ref", couplings.alphas.value)
+        positive("alpha_em_ref", couplings.alphaem.value)
+        positive("scale_ref", couplings.alphas.scale)
         if order[0] not in [0, 1, 2, 3, 4]:
             raise NotImplementedError("a_s beyond N3LO is not implemented")
         if order[1] not in [0, 1, 2]:
             raise NotImplementedError("a_em beyond NLO is not implemented")
         self.order = tuple(order)
+        method = method.value
         if method not in ["expanded", "exact"]:
```

Comment:
Can this still fail if we have an enum already?

Reply:
I applied "minimal" changes, as much as possible. In principle, you can pass whatever at runtime, the check about the type is only in the static analyzer, so in a sense the two tests are decoupled. But if you pass the static analyzer, in principle should be enough.